src/main/java/edu/uw/edm/edmzuulproxy/healthcheck/GwsHealthIndicator.java

```diff
@@ -0,0 +1,39 @@
+package edu.uw.edm.edmzuulproxy.healthcheck;
+
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+import edu.uw.edm.gws.GroupsWebServiceClient;
+import edu.uw.edm.gws.model.GWSSearchType;
+import edu.uw.edm.gws.model.GroupReference;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class GwsHealthIndicator implements HealthIndicator {
+
+    private GroupsWebServiceClient groupsWebServiceClient;
+
+    @Autowired
+    public GwsHealthIndicator(GroupsWebServiceClient groupsWebServiceClient) {
+        this.groupsWebServiceClient = groupsWebServiceClient;
+    }
+
+    @Override
+    public Health health() {
+        String username = "edmsci"; // hardcode app user name for simplicity
+        try {
+            List<GroupReference> groupsForUser = groupsWebServiceClient.getGroupsForUser(username, GWSSearchType.direct);
```

Comment:
What happens if the GWS url changes? if they retire "group_sws/v2/search" and we need to upgrade to v3, will zuul return down?

Reply:
I just tested locally, by using a locally built gws-client with an invalid URL. The indicator was "Down" with  a "Not Found" exception:
2020-02-24 14:41:29.167 ERROR 72140 --- [-169.254.195.29] e.u.e.e.healthcheck.GwsHealthIndicator   : exception calling groupsWebServiceClient.getGroupsForUser(edmsci): 404 Not Found

org.springframework.web.client.HttpClientErrorException: 404 Not Found
	at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:118) ~[spring-web-5.0.13.RELEASE.jar:5.0.13.RELEASE]
	at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:103) ~[spring-web-5.0.13.RELEASE.jar:5.0.13.RELEASE]